core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -2560,20 +2560,94 @@ wasm_interp_dump_call_stack(struct WASMExecEnv *exec_env)
             }
         }
 
+        frame.func_name_wp = func_name;
+
+        bh_vector_append(module_inst->frames, &frame);
+
+        cur_frame = cur_frame->prev_frame;
+        n++;
+    }
+
+    return true;
+}
+
+#define PRINT_OR_DUMP()                                                \
+    do {                                                               \
+        if (print) {                                                   \
+            os_printf("%s", line_buf);                                 \
+        }                                                              \
+        else if (buf) {                                                \
+            uint32 remain_len = len - total_len;                       \
+            uint32 string_len =                                        \
+                snprintf(buf + total_len, remain_len, "%s", line_buf); \
+            if (string_len >= remain_len) {                            \
+                /* Buffer full */                                      \
+                return len;                                            \
+            }                                                          \
+            total_len += string_len;                                   \
+        }                                                              \
+        else {                                                         \
+            total_len += strlen(line_buf);                             \
+        }                                                              \
+    } while (0)
```

Comment:
Same as above, how about make it as function?

Reply:
Done, thanks~